### PR TITLE
fix(SAI): trigger diff query when generating comments + refactor

### LINKED ISSUE #2467: SAI Comments not triggering a diff
If you update a SAI script with only Generate comments it will not provide a new diff and the new comment will not be updated. You need to change an actual param for the comment to update in the diff

### PR DESCRIPTION
**Pull request by Helias**
- Closes #2467
- closes https://github.com/azerothcore/Keira3/issues/2174

Hope this fix does not break https://github.com/azerothcore/Keira3/issues/1820, @Gultask if you can test would be much appreciated, I tested it locally and I could not reproduce the issue, but it's better a double check for this

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -42,6 +42,7 @@ apps/keira/coverage
 npm-debug.log
 testem.log
 /typings
+/src/
```
Comment (FrancescoBorzi) [inline .gitignore:45]: ?

Comment (FrancescoBorzi) [inline .gitignore:45]: ?

**Comment by FrancescoBorzi** [comment]
@Helias what about tests?

**Review by FrancescoBorzi — APPROVED** [review]

**Comment by Gultask** [comment]
It did fix the problem, ty!

**Comment by Gultask** [comment]
doesn't seem to have a problem with LINK events either

**Comment by Gultask** [comment]
Closes https://github.com/azerothcore/Keira3/issues/2174 as well. Also your suggestion of generating comments for a single row only would be highly appreciated because I was actually starting to like having this bug for that reason 😅

**Comment by Helias** [comment]
do you refer to this https://github.com/azerothcore/Keira3/issues/2107 ? I could work on it

thanks for testing, I am merging this PR and closing the two issues, I will prioritize this issue letting the functionality to generate comment only for the selected row

**Pull request merged**